mypy/server/aststripnew.py

```diff
@@ -87,7 +87,9 @@ def prepare_implicit_var_patches(self, node: ClassDef) -> None:
             if isinstance(sym.node, Var) and sym.implicit:
                 explicit_self_type = sym.node.explicit_self_type
 
-                def patch() -> None:
+                # These arguments should not be passed, we just want to capture
+                # the names in closure at current iteration in the for-loop.
+                def patch(name: str = name, sym: SymbolTableNode = sym) -> None:
```

Comment:
This is tragically a bug under mypyc, which doesn't properly bind default variables at function creation time (https://github.com/mypyc/mypyc/issues/336).

There is a bad workaround for this, which is to have another nested function take them as arguments and have them get bound that way. I thought that I had needed to do this somewhere in mypy but I can't find it.

(As an aside: I noticed this because mypyc generates a warning on it, and it is now the only warning, so I actually noticed it.)